src/requests/hover.jl

```diff
@@ -66,12 +66,45 @@ function get_hover(b::StaticLint.Binding, documentation::String, server)
     return documentation
 end
 
+# print(io, x::SymStore) methods are defined in SymbolServer
 function get_hover(b::SymbolServer.SymStore, documentation::String, server)
     if !isempty(b.doc)
         documentation = string(documentation, b.doc, "\n")
     end
     documentation = string(documentation, "```julia\n", b, "\n```")
-    # print(io, x::SymStore) methods are defined in SymbolServer
+end
+
+const JULIA_DIR = normpath(joinpath(Sys.BINDIR, Base.DATAROOTDIR, "julia"))
+
+function get_hover(f::SymbolServer.FunctionStore, documentation::String, server)
+    if !isempty(f.doc)
+        documentation = string(documentation, f.doc, "\n")
+    end
+
+    documentation = string(documentation, "`$(f.name)` is a `Function`.\n")
+    nm = length(f.methods)
+    documentation = string(documentation, "**$(nm)** method", nm == 1 ? "" : "s", " for function ", '`', f.name, '`', '\n')
+    for m in f.methods
+        io = IOBuffer()
+        print(io, m.name, "(")
+        nsig = length(m.sig)
+        for (i,sig) = enumerate(m.sig)
+            if sig[1] ≠ Symbol("#unused#")
+                print(io, sig[1])
+            end
+            print(io, "::", sig[2])
+            i ≠ nsig && print(io, ", ")
+        end
+        print(io, ")")
+        sig = String(take!(io))
+        mod = m.mod
+        text = string(replace(m.file, SymbolServer.JULIA_DIR => ""), ':', m.line)
```

Comment:
I think this won't work with the statically compiled LS server: it is accessing the JULIA_DIR var in the LS process, right?

The key thing is that we _only_ use the dir for base that we get from querying `Sys.BINDIR` and `Base.DATAROOTDIR` in the symbol server process.

Reply:
still not sure how I can solve what you're concerning.
After all, (statically compiled) LS process can't know which version of Julia an user wants to interacts with, no ?

Reply:
couple of questions:
- will SymbolServer.jl be statically compiled ?
- if SymbolServer compiled, and it will replace its `JULIA_DIR` at runtime (i.e. given based on user-configured julia), then `SymbolServer.JULIA_DIR` here won't point at that ? Will static compiler will resolve constants in other modules at compile time ?

If questions above are true, I agree that the easiest way to implement this would be replace this kind of stuff in SymbolServer, and keep that in additional `MethodStore`'s store.
But imho the field doesn't deserve its complexity (non-Base functions don't need that after all), so I'm totally okay to remove these replacements from the beginning. It's just that it will just look verbose.

Reply:
> will SymbolServer.jl be statically compiled ?

The part in SymbolServer.jl that is running inside the LS process will be statically compiled, but the symbol server process won't, we will just continue to start a Julia instance for that.

Re additional stuff in the `MethodStore`, I'm not sure, probably @ZacLN has a better sense of that.

Reply:
for now, I just remove this replacement.

Reply:
Sounds good!